### PR TITLE
remove simple page in favour of sonata's simple pager

### LINKED ISSUE #320: Remove duplicate code in SimplePager
In PR sonata-project/SonataAdminBundle#2656 I have copied the SimplePager from this bundle to SonataAdminBundle as it is very nice feature that should be available also for other users. SimplePager will be available in SonataAdmin 2.4.

To have less code duplicated after the SonataAdmin 2.4 will be released I suggest to remove duplicate code in SonataDoctrinePhpcrAdminBundle's SimplePager and extend the class from SonataAdminBundle's SimplePager. This will be a BC change.

``` php
namespace Sonata\DoctrinePHPCRAdminBundle\Datagrid;

use Sonata\AdminBundle\Datagrid\SimplePager as BaseSimplePager;

class SimplePager extends BaseSimplePager
{
...
}
```

Maybe we can also mark the SonataDoctrinePhpcrAdminBundle's SimplePager as deprecated if the code of SimplePager will be same after the SonataAdmin's release.

### PR DESCRIPTION
**Pull request by ElectricMaxxx**
fix #320 

## Changelog

```markdown
**2017-02-03**: [BC break] Removed own SimplePager in favour of `Sonata\AdminBundle\Datagrid\SimplePager` to avoid duplications.
```

- [x] Update the tests
- [x] Add an upgrade note

## Subject

As described in the Issue, the `SimplePager` was copied into the main, bundle. So we start using the new on here.

### CODE REVIEW

**Comment by dbu** [comment]
cool if thats enough to fix this!

oh well. looks like the build of this repo is quite broken. did you branch from master or from 1.x? we should try to fix the build in a separate branch.

**Comment by ElectricMaxxx** [comment]
@dbu I branched from master.

**Comment by ElectricMaxxx** [comment]
uh @SonataCI  moves my ticket on the board by removeing a label?

**Comment by ElectricMaxxx** [comment]
@dbu i created #417 for the build and php versions.

**Comment by Soullivaneuh** [comment]
@ElectricMaxxx Sonata project labels are commons and managed by dev-kit only.

**Comment by ElectricMaxxx** [comment]
I just was confused. I use waffle.io and while editing an other issue,
this one was moving :-)

**Comment by ElectricMaxxx** [comment]
@dbu  tested it on sandbox, works fine.

**Comment by greg0ire** [comment]
I restarted the build, it should be better now.

**Comment by greg0ire** [comment]
Hmm, it's not better. I think I have to close and reopen.

**Comment by greg0ire** [comment]
Nope… still not good.

**Comment by greg0ire** [comment]
Oh I get it, your change is on master, which does not have the fix yet.

File: Datagrid/SimplePager.php [hunk 1]
```diff
@@ -1,159 +0,0 @@
-<?php
```
Comment (greg0ire) [inline Datagrid/SimplePager.php]: Can you consider deprecating this on the stable branch first?

**Comment by ElectricMaxxx** [comment]
The deprecation should be done on a 1.x branch, right? doing it on
master and merging first would not make any sense.

**Comment by greg0ire** [comment]
"stable branch" == 1.x, so yeah.

**Comment by greg0ire** [comment]
If you intend to do so, please change the base branch of this PR, and then rebase to get rid of the commits that are not yours.

**Comment by greg0ire** [comment]
> fix #320

Does it? #320 is a PR, which you should still leave open and have merged once the stable branch is merged in the unstable branch.

**Comment by ElectricMaxxx** [comment]
For sure and i will remove the test then and trust yours :-)

**Comment by greg0ire** [comment]
Sorry, I moved a message to the right PR, not sure what you're answering to anymore 😅

**Comment by dbu** [comment]
okay, the deprecation is now in master. which leads to a conflict :-/

@ElectricMaxxx can you please rebase this PR?

**Comment by ElectricMaxxx** [comment]
I did so. lets see what is happening.

File: Datagrid/SimplePager.php (hunk 1 above)
Comment (dbu) [inline Datagrid/SimplePager.php]: done

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 Changelog
 =========
 
+* **2017-02-03**: [BC break] Removed own SimplePager in favour of `Sonata\AdminBundle\Datagrid\SimplePager` to avoid duplications.
```
Comment (greg0ire) [inline CHANGELOG.md]: You're not really supposed to touch this, that's what the changelog in the PR body is about. Please remove this and make the changelog PR as good as this one.

**Comment by greg0ire** [comment]
@dbu please don't approve until the todo list is complete, otherwise we might merge this by mistake.

**Comment by ElectricMaxxx** [comment]
i didn't checked the items as i didn't expect do do somethin on that points.

**Comment by greg0ire** [comment]
> i didn't checked the items as i didn't expect do do somethin on that points.

Well then remove the items. We should change our PR template, I think no one understands it.

**Comment by greg0ire** [comment]
But I think there should be an upgrade note anyway, right?

File: CHANGELOG.md (hunk 2 above)
Comment (ElectricMaxxx) [inline CHANGELOG.md]: This is working automaticially too?
whoot? never seen that.

Comment (greg0ire) [inline CHANGELOG.md]: It's not automatic, it's a human ( @Soullivaneuh ) that does this. The goal is to avoid n PRs conflicting with each other.

**Comment by ElectricMaxxx** [comment]
@greg0ire i did it, can i squash?

To your template:
yes the amount of text and posibilities avoids to find the relevant data. It looks like a 10 page form for my taxes, and  when a PR has an entry point like doing taxes ...

**Comment by greg0ire** [comment]
> @greg0ire i did it, can i squash?

Can and should

For the PR template, I'll try to work on simplifying it.

**Comment by ElectricMaxxx** [comment]
looks fine from me. Ready to merge.

**Comment by greg0ire** [comment]
Thanks @ElectricMaxxx , I was waiting for the build.

**Comment by greg0ire** [comment]
Please use the same format we use in every other PR for your CHANGELOG. See the PR template.

**Comment by greg0ire** [comment]
BTW @dbu why is this particular repo using a different format for its changelog?

**Comment by dbu** [comment]
> BTW @dbu <https://github.com/dbu> why is this particular repo using a
> different format for its changelog?

probably historical reasons. sonata was not that formalized earlier on,
and the phpcr admin lived a bit outside the core sonata team :-)

will your automated thing work if we leave the older changelog as is?

also, is there maintainer documentation on how to release?

**Comment by greg0ire** [comment]
> will your automated thing work if we leave the older changelog as is?

"our automated thing" is @Soullivaneuh , so yeah, it should work :smile: 

> also, is there maintainer documentation on how to release?
Not yet, but I wrote [this PR](https://github.com/sonata-project/dev-kit/pull/227). Please review it.

**Pull request merged**